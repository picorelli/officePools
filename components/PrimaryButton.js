import React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { colors, padding } from '../styles/base'

const PrimaryButton = props => (
  <TouchableOpacity
    {...props}
    style={{
      backgroundColor: colors.secondary,
      paddingVertical: padding.md,
    }}
  >
    <Text
      style={{
        color: colors.primary,
        textAlign: 'center',
        fontWeight: '700',
        fontSize: 16,
      }}
    >
      {props.label}
    </Text>
  </TouchableOpacity>
)

export default PrimaryButton
