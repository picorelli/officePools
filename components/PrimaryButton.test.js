import React from 'react'
import renderer from 'react-test-renderer'
import PrimaryButton from './PrimaryButton'

it('renders without crashing', () => {
  const rendered = renderer.create(<PrimaryButton />).toJSON()

  expect(rendered).toBeTruthy()
})

it('renders correctly', () => {
  const tree = renderer.create(<PrimaryButton />).toJSON()

  expect(tree).toMatchSnapshot()
})
