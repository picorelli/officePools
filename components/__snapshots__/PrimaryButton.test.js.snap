// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "opacity": 1,
      "paddingVertical": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgba(224, 31, 35, 1)",
        "fontSize": 16,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
  />
</View>
`;
